client/src/js/app/tasks.filter.js

```diff
@@ -0,0 +1,28 @@
+angular.module('toDoList').filter('tasksFilter', tasksFilter);
+
+function tasksFilter() {
+  return function (input, filterBy) {
+    var out = [];
+    filterBy = filterBy.toLowerCase();
+    if (filterBy == "all") {
+      out = input;
+    } else {
+      angular.forEach(input, function (task) {
```

Comment:
could you please check out http://underscorejs.org and use it here (`_.map` or `_.findWhere`)?